packages/node-experimental/src/utils/prepareEvent.ts

```diff
@@ -0,0 +1,58 @@
+import { Scope } from '@sentry/core';
+import type { CaptureContext, EventHint, Scope as ScopeInterface, ScopeContext } from '@sentry/types';
+
+/**
+ * This type makes sure that we get either a CaptureContext, OR an EventHint.
+ * It does not allow mixing them, which could lead to unexpected outcomes, e.g. this is disallowed:
+ * { user: { id: '123' }, mechanism: { handled: false } }
+ */
+export type ExclusiveEventHintOrCaptureContext =
+  | (CaptureContext & Partial<{ [key in keyof EventHint]: never }>)
+  | (EventHint & Partial<{ [key in keyof ScopeContext]: never }>);
+
+/**
+ * Parse either an `EventHint` directly, or convert a `CaptureContext` to an `EventHint`.
+ * This is used to allow to update method signatures that used to accept a `CaptureContext` but should now accept an `EventHint`.
+ */
+export function parseEventHintOrCaptureContext(
+  hint: ExclusiveEventHintOrCaptureContext | undefined,
+): EventHint | undefined {
+  if (!hint) {
+    return undefined;
+  }
+
+  // If you pass a Scope or `() => Scope` as CaptureContext, we just return this as captureContext
+  if (hintIsScopeOrFunction(hint)) {
+    return { captureContext: hint };
+  }
+
+  if (hintIsScopeContext(hint)) {
+    return {
+      captureContext: hint,
+    };
+  }
+
+  return hint;
+}
```

Comment:
This is messed up but I don't know what to do about it. Is captureException api just garbage?

Reply:
Yeah this is because in the global `Sentry.captureException()` the API used to be `captureException(exception, captureContext)`, while in other places the second argument is `EventHint`. 

We could deprecate this for v8 and require this to be a hint, which means that users need to refactor:

```js
Sentry.captureException(error, { captureContext: { tags: ... } })
```

🤷